package-res/resources/web/prompting/api/EventAPI.js

```diff
@@ -79,13 +79,31 @@ define([], function() {
      * @method
      * @param {Function} callback   The function to be executed when the event is triggered.
      *                              Pass null if you wish to unbind this event
-     * @example
+     * @param {String} paramName    The name of the parameter to which the callback will be called
+     *                              This argument is optional. If no parameter name is passed on
+     *                              the callback will be assigned to all the parameters.
+     *
+     * @example <caption>Register a parameter changed event to all parameters</caption>
      *  api.event.parameterChanged(function(parameterName, parameterValue) {
      *    // Execute event based code
      *  })
+     *
+     * @example <caption>Register a parameter changed event to the parameter called "parameterName"</caption>
+     *  api.event.parameterChanged(function(parameterName, parameterValue) {
+     *    // Execute event based code
+     *  }, "parameterName")
      */
-    this.parameterChanged = function(callback) {
-      api.operation._getPromptPanel().onParameterChanged = callback;
+    this.parameterChanged = function(callback, paramName) {
+      var onParameterChanged = api.operation._getPromptPanel().onParameterChanged;
+      if (!paramName) {
+        onParameterChanged = callback;
+      } else {
+        if (!onParameterChanged) {
+          onParameterChanged = {};
+        }
+        onParameterChanged[paramName] = callback;
+      }
+      api.operation._getPromptPanel().onParameterChanged = onParameterChanged;
```

Comment:
I think the local variable optimisation here is not working, e.g. if paramName is a falsy the code executes 3 writes (lines 97, 99 and 106) isn't it more efficient to just do the three write cases as such:
```
if (!paramName) {
  api.operation._getPromptPanel().onParameterChanged = callback;
} else {
  if (!api.operation._getPromptPanel().onParameterChanged) {
    api.operation._getPromptPanel().onParameterChanged = {[paramName]: callback};
  } else {
    api.operation._getPromptPanel().onParameterChanged[paramName] = callback;
  }
}
```
or if we cannot use ES6 computed property names:
```
if (!paramName) {
  api.operation._getPromptPanel().onParameterChanged = callback;
} else {
  var onParameterChanged = api.operation._getPromptPanel().onParameterChanged;
  if (!onParameterChanged) {
    onParameterChanged = {};
  }
  onParameterChanged[paramName] = callback;
  api.operation._getPromptPanel().onParameterChanged = onParameterChanged;
}
```
Also, if we execute the first example in the documentation and then execute the second example (we also want to execute a more specific callback for a parameter "A" changed event). If we do this in the inverse order (first execute example 2 and then execute example 1) do we want to lose the (parameter-specific) callback assigned in example 2? e.g.:
```
it("should allow to assign an overall callback and also a callback for a specific parameter", function() {
  var parameterCallback1 = function() {};
  var parameterCallback2 = function() {};
  eventApi.parameterChanged(parameterCallback1, 'param');
  eventApi.parameterChanged(parameterCallback2);
  expect(promptPanelSpy.onParameterChanged).toBeDefined();
  expect(promptPanelSpy.onParameterChanged['param']).toBe(parameterCallback1);
  expect(promptPanelSpy.onParameterChanged).toBe(parameterCallback2);
});
```
although this works (```eventApi.parameterChanged(parameterCallback2);``` executes first):
```
it("should allow to assign an overall callback and also a callback for a specific parameter", function() {
  var parameterCallback1 = function() {};
  var parameterCallback2 = function() {};
  eventApi.parameterChanged(parameterCallback2);
  eventApi.parameterChanged(parameterCallback1, 'param');
  expect(promptPanelSpy.onParameterChanged).toBeDefined();
  expect(promptPanelSpy.onParameterChanged['param']).toBe(parameterCallback1);
  expect(promptPanelSpy.onParameterChanged).toBe(parameterCallback2);
});
```